### ISSUE TITLE
implement HttpResponseStatusCode to string

### ISSUE THREAD

**Issue by tigercosmos** [seladb/PcapPlusPlus#1157]
there is no `toString()` function for `HttpResponseStatusCode`

it would be nice to have:

```cpp
HttpResponseStatusCode code = HttpResponseStatusCode::Http200OK;
code.toString(); // "HTTP 200 OK"
```

We already have something like `StatusCodeEnumToString`, however it's only in cpp code and the messages are not clear enough, e.g. `Http200OK` -> `OK` but it's better to have `HTTP 200 OK`

**Comment by seladb** [comment]
`HttpResponseStatusCode` is an enum so if cannot have `toString()`...

However there is already such method: `getStatusCodeString()`. Here is how to use it:

```cpp
pcpp::HttpResponseLayer* responseLayer = httpPacket.getLayerOfType<pcpp::HttpResponseLayer>();
std::cout << responseLayer->getFirstLine()->getStatusCodeString() << std::endl;

// "OK" will be printed
```

**Comment by tigercosmos** [comment]
Thanks for your reply. Currently, `getStatusCodeAsInt` prints the code in `HttpResponseFirstLine`.
When there is a standalone status code, there is no function to print it out.

here is the use case:

we probably can have a method like 
```cpp
static HttpResponseLayer::HttpResponseStatusCode HttpResponseLayer::getStatusCodeAsInt(HttpResponseLayer::HttpResponseStatusCode code);
```

```cpp
pcpp::HttpResponseLayer* responseLayer = httpPacket.getLayerOfType<pcpp::HttpResponseLayer>();
if(responseLayer->getStatusCode()  != HttpResponseStatusCode::Http200OK) {
   std::cerr << "got status code: " << responseLayer->getFirstLine()->getStatusCodeAsInt()
             << ", expect status code: " << HttpResponseLayer::getStatusCodeAsInt(HttpResponseStatusCode::Http200OK);
}
```

**Comment by tigercosmos** [comment]
@seladb  any comments?

**Comment by seladb** [comment]
@tigercosmos I'm sorry for the late reply

Maybe we can just add these status non-standard status codes? 🤔

**Comment by tigercosmos** [comment]
@seladb  sorry, I don't understand, could you provide an example?

**Comment by seladb** [comment]
sorry @tigercosmos I misread your previous comments.
If we want to add `toString()` to the `HttpResponseStatusCode` enum we probably need to create a wrapper class for it, similar to the suggestion here: https://stackoverflow.com/a/53284026

I think it's doable and will add some value. It can also use the existing structures in `HttpLayer.cpp` such as `StatusCodeEnumToString` and `StatusCodeEnumToInt`. Would you consider implementing it?

**Comment by tigercosmos** [comment]
sure, sounds interesting, let me do it

**Comment by seladb** [comment]
@tigercosmos are you still planning to work on it?

**Comment by tigercosmos** [comment]
@seladb  yes, sorry for the late. I will finish it in these days.

**Comment by seladb** [comment]
Fixed by #1180 . Thank you so much @tigercosmos for adding this to PcapPlusPlus! 🙏
